bot/src/discord/schedule_helper.js

```diff
@@ -0,0 +1,306 @@
+const dbConnection = require('kotoba-node-common').database.connection;
+const WordScheduleModel = require('kotoba-node-common').models.createWordSchedulesModel(dbConnection);
+const showRandomWord = require('./show_random_word.js');
+
+const frequencyCheck = 60 * 1000;
+
+const statusConstants = {
+  running: 'running',
+  paused: 'paused',
+};
+
+/**
+ * Starts the interval
+ * @param {Monochrome} monochrome
+ * @param {boolean} firstCall
+ */
+async function setTimer(monochrome, firstCall) {
+  let now = (new Date()).getTime();
+  // The first time this function is called we don't want to run the command,
+  // just set the timer so it runs in sync with the frequencyCheck value
+  if (!firstCall) {
+    const pendingSchedules = await WordScheduleModel.find({ status: statusConstants.running });
+    pendingSchedules.forEach(async (schedule) => {
+      // We check if the schedule has reached the start date and if it's in sync
+      // with the frequency, with an error margin of 0.1s
+      if (now >= schedule.start && (now - schedule.start) % schedule.frequency < 100) {
+        const channel = monochrome.getErisBot()
+          .guilds.get(schedule.serverId)
+          .channels.get(schedule.id);
```

Comment:
We'll need to handle if Kotoba can't send a message to this channel. We could add a function like:

```js
// Returns the channel for this WOTD schedule, or undefined
// if we cannot get the channel or cannot send messages in it.
getWOTDChannel(eris, serverId, channelId) {
  const server = eris.guilds.get(serverId);

  // If Kotoba is no longer in the server, return undefined.
  if (!server) {
    return undefined;
  }

  const channel = server.channels.get(channelId);
  
  // If it's a guild channel, verify correct permissions.
  if (channel.guild) {
    // This checks that the channel is a TextChannel. Only they have this function. 
    if (!channel.permissionsOf) {
      return undefined;
    }

    // Permissions that Kotoba has in this channel
    const permissions = channel.permissionsOf(eris.user.id);

    // Make sure we can send messages
    if (!permissions.has('sendMessages')) {
      return undefined;
    }
  }

  // All good
  return channel;
}
```
Then you can use that function to get the channel, and if it's undefined, you can delete the schedule because Kotoba is no longer able to send messages to that channel.

Reply:
Isn't it too drastic to delete it? Maybe the admin is tweaking the permissions and after a while the bot can write again in the channel. Well, it's not like they would lose a lot of work anyway.

Reply:
We can leave it alone then, if we record the lastSent in the database. That would allow me to manually purge all schedules that haven't been sent in a long time, if I find there is a proliferation of dead schedules.